### PR TITLE
Fix TextButtons being displayed in black

### LINKED ISSUE #1077: In dark theme, change-server and opt-out-of-analytics buttons are black-on-black
### Steps to reproduce

1. Launch app on dark theme
2. Fail to find button to change server
3. Realise it's black on black

#### What did you expect?

Legible buttons

#### What happened instead?

https://www.google.co.uk/books/edition/The_Restaurant_at_the_End_of_the_Univers/r--6ujORIVgC?hl=en&gbpv=1&pg=PA121

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

0.1.2 (40001020) playstore

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by jmartinesp**
Set the default `LocalContentColor` to `onSurface`, since it's `Color.Black` by default.

Also modified `WaitListView` to use the right colors, since it's a special case where theme is hardcoded.

Other possible solutions are:
- Set it to `Color.Unspecified` and let the components use a fallback color.
- Ensure there is always a `Surface` component providing this value at the root of each screen.

Fixes #1077.

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.login.impl.screens.waitlistscreen_null_WaitListViewDark_0_null_4,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.login.impl.screens.waitlistscreen_null_WaitListViewDark_0_null_4,NEXUS_5,1.0,en].png:1]: The text is "less" white than before, I do not think this is OK.

image

File: libraries/theme/src/main/kotlin/io/element/android/libraries/theme/ElementTheme.kt [hunk 1]
```diff
@@ -115,6 +116,7 @@ fun ElementTheme(
     }
     CompositionLocalProvider(
         LocalCompoundColors provides currentCompoundColor,
+        LocalContentColor provides colorScheme.onSurface,
```
Comment (bmarty) [inline libraries/theme/src/main/kotlin/io/element/android/libraries/theme/ElementTheme.kt:119]: Should it be `colorScheme.primary` instead?

**Review by bmarty — COMMENTED** [review]
Thanks, a remark.

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.login.impl.screens.waitlistscreen_null_WaitListViewDark_0_null_4,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.login.impl.screens.waitlistscreen_null_WaitListViewDark_0_null_4,NEXUS_5,1.0,en].png:1]: It's actually matching the design in Figma (if I'm looking at the right one). The title uses `textPrimary` and the subtitle uses pure white for some reason.

File: libraries/theme/src/main/kotlin/io/element/android/libraries/theme/ElementTheme.kt (hunk 1 above)
Comment (jmartinesp) [reply libraries/theme/src/main/kotlin/io/element/android/libraries/theme/ElementTheme.kt:119]: Both semantic tokens point to the same color token, and I think `onSurface` matches the default behavior better, as this `LocalContentColor` is usually provided by a parent `Surface` component, but I can change it if needed.

**Comment by bmarty** [comment]
(just to make this public: subtitle as to use the primary color as well. Figma has been updated)

**Comment by jmartinesp** [comment]
> (just to make this public: subtitle as to use the primary color as well. Figma has been updated)

This should be fixed now.

**Review by bmarty — APPROVED** [review]
Thanks!

**Pull request merged**